toonz/sources/toonzqt/doublefield.cpp

```diff
@@ -461,6 +494,58 @@ void MeasuredDoubleLineEdit::timerEvent(QTimerEvent *) {
   }
 }
 
+//-----------------------------------------------------------------------------
+
+void MeasuredDoubleLineEdit::mousePressEvent(QMouseEvent *e) {
+  if ((e->buttons() == Qt::MiddleButton) || m_labelClicked) {
+    m_xMouse           = e->x();
+    m_mouseDragEditing = true;
+  } else
+    QLineEdit::mousePressEvent(e);
+}
+
+//-----------------------------------------------------------------------------
+
+void MeasuredDoubleLineEdit::mouseMoveEvent(QMouseEvent *e) {
+  if ((e->buttons() == Qt::MiddleButton) || m_labelClicked) {
+    m_value->modifyValue((e->x() - m_xMouse) / 2);
```

Comment:
In some Fx there are parameters with small value range. For instance, the `Opacity` parameter of the `Over Ino Fx` has 0.0-1.0 value range. In such case middle-dragging of the field cannot get the values other than 0 and 1 for now.

How about modifying this line, adding the `precision` argument like as follows ?:
```
 int precision = (m_maxValue > 100) ? 0 : ((m_maxValue > 10) ? 1 : 2);
 m_value->modifyValue((e->x() - m_xMouse) / 2, precision);
```
I once thought of referring `MeasuredDoubleLineEdit::m_decimals` variable but using the maximum value of the field seems to perform better.